systems/plants/collisionDetectmex.cpp

```diff
@@ -33,83 +30,65 @@ void mexFunction( int nlhs, mxArray *plhs[],int nrhs, const mxArray *prhs[] ) {
-    //END_DEBUG
-    int n_active_bodies = static_cast<int>(mxGetNumberOfElements(body_idx));
-    //DEBUG
-    //cout << "collisionDetectmex: n_active_bodies = " << n_active_bodies << endl;
-    //END_DEBUG
+    size_t n_active_bodies = static_cast<size_t>(mxGetNumberOfElements(body_idx));
+
     active_bodies_idx.resize(n_active_bodies);
     memcpy(active_bodies_idx.data(),(int*) mxGetData(body_idx),
            sizeof(int)*n_active_bodies);
-    transform(active_bodies_idx.begin(),active_bodies_idx.end(),
+    transform(active_bodies_idx.begin(), active_bodies_idx.end(),
               active_bodies_idx.begin(),
               [](int i){return --i;});
   }
-  // DEBUG
-  //cout << "collisionDetectmex: Parse active_bodies END" << endl;
-  // END_DEBUG
 
-  // Then get the group names for which to compute distances
-  // DEBUG
-  //cout << "collisionDetectmex: Parse collision groups START" << endl;
-  // END_DEBUG
-  const mxArray* collision_groups = mxGetField(active_collision_options,0,
-                                               "collision_groups");
+  const mxArray* collision_groups = mxGetField(active_collision_options, 0, "collision_groups");
+  
   if (collision_groups != NULL) {
-    int num = static_cast<int>(mxGetNumberOfElements(collision_groups));
-    for (int i=0; i<num; i++) {
-      const mxArray *ptr = mxGetCell(collision_groups,i);
-      int buflen = static_cast<int>(mxGetN(ptr)*sizeof(mxChar))+1;
+    size_t num_collision_groups = static_cast<size_t>(mxGetNumberOfElements(collision_groups));
+    for (size_t i = 0; i < num_collision_groups; i++) {
+      const mxArray *ptr = mxGetCell(collision_groups, i);
+      size_t buflen = static_cast<size_t>(mxGetN(ptr) * sizeof(mxChar)) + 1;
       char* str = (char*)mxMalloc(buflen);
       mxGetString(ptr, str, buflen);
       active_group_names.insert(str);
       mxFree(str);
     }
   }
-  // DEBUG
-  //cout << "collisionDetectmex: Parse collision groups END" << endl;
-  // END_DEBUG
 
   vector<int> bodyA_idx, bodyB_idx;
   MatrixXd ptsA, ptsB, normals, JA, JB, Jd;
   VectorXd dist;
-  // DEBUG
-  //cout << "collisionDetectmex: Call collisionDetect START" << endl;
-  // END_DEBUG
+
   if (active_bodies_idx.size() > 0) {
     if (active_group_names.size() > 0) {
-      model-> collisionDetect(dist, normals, ptsA, ptsB, bodyA_idx, bodyB_idx,
+      model->collisionDetect(dist, normals, ptsA, ptsB, bodyA_idx, bodyB_idx,
                               active_bodies_idx,active_group_names);
     } else {
-      model-> collisionDetect(dist, normals, ptsA, ptsB, bodyA_idx, bodyB_idx,
+      model->collisionDetect(dist, normals, ptsA, ptsB, bodyA_idx, bodyB_idx,
                               active_bodies_idx);
     }
   } else {
-    if (active_group_names.size() > 0) {
-      model-> collisionDetect(dist, normals, ptsA, ptsB, bodyA_idx, bodyB_idx,
+    const bool multiple_contacts = mxIsLogicalScalarTrue(allow_multiple_contacts);
```

Comment:
is this doing sufficient argument checking?  what happens if allow_multiple_contacts is a double?  or is NULL?